### ISSUE TITLE
Ability to specify a target PhaseType when changing Plan

### ISSUE THREAD

**Issue by sbrossie** [killbill/killbill#704]
Today, we can specify the `changeAlignment` policy in the catalog to impact which `PhaseType` will be chosen (and how far in that phase we end up) when we make the change operation.

However there are situations, [as described in this test scenario](https://github.com/killbill/killbill-integration-tests/blob/master/killbill-integration-tests/core/test_entitlement_change_skip_phase.rb#L69) where this is not enough and we want the ability to specify the target `PhaseType`

**Comment by sbrossie** [comment]
**Description of the API**

We have reused the structure that is currently used to override catalog pricing when creating a `Subscription` (or changing `Plan`). The `priceOverrides` section **must contain** only one element and only the `phaseType` must be specified (everything else should be left unspecified or null.

**Example**

An `CURL` example to swicth to a `Gold` `Plan` with two phases `{TRIAL, EVERGREEN}` and directly land on the `EVERGREEN` phase would look like the following:

```
curl -v \
     -u admin:password \
     -H "X-Killbill-ApiKey: bob" \
     -H "X-Killbill-ApiSecret: lazar" \
     -H "Content-Type: application/json" \
     -H "X-Killbill-CreatedBy: demo" \
     -X PUT \
     --data-binary '{"accountId":"2f2cd35d-f837-4cff-b547-59329e9b2abb", "planName":"gold-monthly", "productCategory": "BASE", "priceOverrides": [ {"phaseType": "EVERGREEN"} ] }' \
     "http://127.0.0.1:8080/1.0/kb/subscriptions/9d569cbe-140c-414b-9810-189403d71f27"
```

**Comment by sbrossie** [comment]
Also for more information on the behavior of the system when using catalog plan alignment policies along with ability to skip phases, please refer to this [doc](http://docs.killbill.io/latest/plan_alignment.html)

**Comment by sbrossie** [comment]
Workaround has been implemented in `0.18.4` as described [above](https://github.com/killbill/killbill/issues/704#issuecomment-277770707)

Re-opening this issue to provide a real descent api to cover that use case in `0.20`.
